### PR TITLE
Improve logging for dynamic config

### PR DESCRIPTION
**Pull request by longquanzheng**
**What changed?**
Change the Warn from Info for key not found when reading dynamic config

**Why?**
It's noisy to report the warnings. And it's not accurate to use Warn for the case that config is not set. Afterall using default values is expected to do. 

**How did you test it?**
Existing tests.

**Potential risks**
No.

### CODE REVIEW

File: common/service/dynamicconfig/config.go [hunk 1]
```diff
@@ -26,6 +26,8 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/uber/cadence/.gen/go/shared"
+
```
Comment (yycptt) [inline common/service/dynamicconfig/config.go:30]: nit: remove empty line and use internal types

File: common/service/dynamicconfig/clientInterface.go [hunk 2]
```diff
@@ -45,3 +47,7 @@ type Client interface {
 	// UpdateValue takes value as map and updates by overriding. It doesn't support update with filters.
 	UpdateValue(name Key, value interface{}) error
 }
+
+var notFoundError = &shared.EntityNotExistsError{
```
Comment (yycptt) [inline common/service/dynamicconfig/clientInterface.go:51]: As we are migrating to internal types. Can we use types defined in `common/types` instead?

Comment (longquanzheng) [reply common/service/dynamicconfig/clientInterface.go:51]: Yeah good point. almost forgot that.

File: common/service/dynamicconfig/clientInterface.go [hunk 3]
```diff
@@ -23,6 +23,7 @@
 package dynamicconfig
 
 import (
+	"github.com/uber/cadence/common/types"
 	"time"
```
Comment (yycptt) [inline common/service/dynamicconfig/clientInterface.go:27]: nit: `make fmt`

**Review by yycptt — APPROVED** [review]

**Pull request merged**